.github/workflows/publish-sdk-go.yml

```diff
@@ -0,0 +1,26 @@
+name: publish-sdk-go
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - sdk/go/**
+    paths:
+      - "sdk/go/**"
```

Comment:
Just wanted to share context [not important, not urgent, debatable]:

I've been trying to part away from `paths` based filters, because:

- Ideally, with dagger, they're not needed (see self-hosted PR where irrelevant jobs run in ~2-3 seconds)
- local/ci divergence: not the same behavior if you run locally
- Can't enable GH enforcement since they conditionally come up

Reply:
Yes, my thinking is along the same lines. We did the same for the Python SDK: https://github.com/dagger/dagger/pull/3743/files

Will catch this in the next refactoring. Thanks!